src/main/java/seedu/address/timetable/TimetableRetriever.java

```diff
@@ -90,29 +95,45 @@ private static JSONArray getSemesterSpecificTimetableData(int sem, String inline
     }
 
     private static void findLessonAndAdd(List<Lesson> lessons, String module, JSONArray timetableData,
-                                         String[] moduleLessonArray) {
+                                         String[] moduleLessonArray, LocalDate startDate) {
         for (int lessonIter = 0; lessonIter < moduleLessonArray.length; lessonIter++) {
             if (!moduleLessonArray[0].equals("")) {
                 String lessonInfo = moduleLessonArray[lessonIter];
-                addLesson(lessons, module, lessonInfo, timetableData);
+                addLesson(lessons, module, lessonInfo, timetableData, startDate);
             }
         }
     }
 
-    private static void addLesson(List<Lesson> lessons, String module, String lessonInfo, JSONArray timetableData) {
+    private static void addLesson(List<Lesson> lessons, String module, String lessonInfo, JSONArray timetableData,
+                                  LocalDate startDate) {
         String lessonType = getLessonType(lessonInfo);
         String lessonNum = getLessonNum(lessonInfo);
+
         for (int dataIter = 0; dataIter < timetableData.size(); dataIter++) {
             JSONObject currentData = (JSONObject) timetableData.get(dataIter);
             String currentLessonType = (String) currentData.get("lessonType");
             String currentLessonNum = (String) currentData.get("classNo");
+
             if (currentLessonType.equals(lessonType) && currentLessonNum.equals(lessonNum)) {
                 String day = (String) currentData.get("day");
-                String startTime = DUMMY_DATE + (String) currentData.get("startTime");
-                String endTime = DUMMY_DATE + (String) currentData.get("endTime");
-                String name = module + currentLessonType;
-                lessons.add(new Lesson(new Name(name), new Deadline(startTime), new Deadline(endTime),
-                        new ModuleCode(module)));
+                String name = module + " " + currentLessonType + " " + day;
+
+                JSONArray jsonWeeks = (JSONArray) currentData.get("weeks");
+                int[] weeks = getWeeksAsIntegerArray(jsonWeeks);
+                int dayOffset = getDayOffset(day);
+
+                for (int week : weeks) {
+                    int daysToAdd = 7 * (week - 1) + dayOffset;
+                    LocalDate date = startDate.plusDays(daysToAdd);
+                    if (date.isAfter(LocalDate.now())) {
+                        String startTime = date.format(LOCAL_DATE_FORMATTER) + " "
+                                + (String) currentData.get("startTime");
+                        String endTime = date.format(LOCAL_DATE_FORMATTER) + " "
+                                + (String) currentData.get("endTime");
+                        lessons.add(new Lesson(new Name(name), new Deadline(startTime), new Deadline(endTime),
+                                new ModuleCode(module)));
+                    }
+                }
```

Comment:
Good job :) You may want to consider abstracting lines 125 to 135 into another method as this method is quite deeply nested. Remember to SLAP hard :D

Reply:
SLAP-ed. Thank you!

Reply:
👍